web/src/main/java/com/synopsys/integration/alert/web/api/job/JobConfigActions.java

```diff
@@ -230,43 +231,38 @@ public JobConfigActions(AuthorizationManager authorizationManager, DescriptorAcc
-        ValidationResponseModel responseModel = ValidationResponseModel.success("Valid name");
-        for (FieldModel fieldModel : jobFieldModel.getFieldModels()) {
-            responseModel = validateJobNameUnique(currentJobId, fieldModel);
-            if (responseModel.hasErrors()) {
-                return responseModel;
-            }
-        }
-        return responseModel;
+    private Optional<AlertFieldStatus> validateJobNameUnique(@Nullable UUID currentJobId, JobFieldModel jobFieldModel) {
+        Optional<AlertFieldStatus> fieldStatus = jobFieldModel.getFieldModels().stream()
+                                                     .filter(fieldModel -> fieldModel.getFieldValueModel(ChannelDistributionUIConfig.KEY_NAME).isPresent())
+                                                     .findFirst()
+                                                     .flatMap(fieldModel -> fieldModel.getFieldValueModel(ChannelDistributionUIConfig.KEY_NAME))
+                                                     .flatMap(fieldValueModel -> validateJobNameUnique(currentJobId, fieldValueModel));
+        return fieldStatus;
     }
 
-    private ValidationResponseModel validateJobNameUnique(@Nullable UUID currentJobId, FieldModel fieldModel) {
-        Optional<FieldValueModel> jobNameFieldOptional = fieldModel.getFieldValueModel(ChannelDistributionUIConfig.KEY_NAME);
+    private Optional<AlertFieldStatus> validateJobNameUnique(@Nullable UUID currentJobId, FieldValueModel fieldValueModel) {
         String error = "";
-        if (jobNameFieldOptional.isPresent()) {
-            String jobName = jobNameFieldOptional.get().getValue().orElse(null);
-            if (StringUtils.isNotBlank(jobName)) {
-                List<ConfigurationJobModel> jobs = configurationAccessor.getAllJobs();
-
-                boolean foundDuplicateName = jobs.stream()
-                                                 .filter(job -> filterOutMatchingJobs(currentJobId, job))
-                                                 .flatMap(job -> job.getCopyOfConfigurations().stream())
-                                                 .map(configurationModel -> configurationModel.getField(ChannelDistributionUIConfig.KEY_NAME).orElse(null))
-                                                 .filter(configurationFieldModel -> (null != configurationFieldModel) && configurationFieldModel.getFieldValue().isPresent())
-                                                 .anyMatch(configurationFieldModel -> jobName.equals(configurationFieldModel.getFieldValue().get()));
-                if (foundDuplicateName) {
-                    error = "A distribution configuration with this name already exists.";
-                }
-            } else {
-                error = "Name cannot be blank.";
+        String jobName = fieldValueModel.getValue().orElse(null);
+        if (StringUtils.isNotBlank(jobName)) {
+            List<ConfigurationJobModel> jobs = configurationAccessor.getAllJobs();
```

Comment:
This is for 6.3.0 right? This method should not be used in 6.4.0+

Reply:
Yes, its for 6.3.0, this is coming from Line 250 in the old code

Reply:
Gotcha. Just make sure when you merge 6.3.0 -> master that you don't use `getAllJobs`.